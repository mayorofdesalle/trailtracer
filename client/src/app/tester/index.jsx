import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
