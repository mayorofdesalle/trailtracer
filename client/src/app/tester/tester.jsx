import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };