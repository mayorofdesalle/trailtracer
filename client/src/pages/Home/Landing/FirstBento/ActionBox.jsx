import { useTranslation } from 'react-i18next';
import styled, { useTheme } from 'styled-components';

import { slideFromLeft, slideFromRight } from '@components/anims';
import BentoBox from '@components/ui/Bento/BentoBox';
import Text from '@components/ui/Text';
import Vector from '@components/ui/Vector';
import useWindowSize from '@hooks/useWindowSize';

/**
 * ActionBox
 * @description
 * Styled bento box to contain the CTA content.
 **/
const ActionBox = styled(BentoBox)`
    grid-area: ${({ $gridArea }) => $gridArea};
    flex-direction: column;
    justify-content: space-around;
    overflow: hidden;

    & > #R {
        animation: ${slideFromRight} 1s ease-in-out;
    }

    & > #L {
        animation: ${slideFromLeft} 1s ease-in-out;
    }
`;

export const ActionBoxFirst = () => {
    const theme = useTheme();
    const { t } = useTranslation();
    const { height, width } = useWindowSize();
    const largeScreen = width > theme.breakpoints.medium && width / height > 1;

    return (
        <ActionBox $gridArea='ActionBox1' $color={theme.colors.primary}>
            <Vector id='L' $name={largeScreen ? 'arrowUpRight' : 'arrowUp'} $color={theme.colors.tertiary} $height='7dvh' $width='7dvh' $position='top' />
            <Text $color={theme.colors.tertiary}>{t('landingPage.CTA1')}</Text>
        </ActionBox>
    );
};

export const ActionBoxSecond = () => {
    const theme = useTheme();
    const { t } = useTranslation();

    return (
        <ActionBox $gridArea='ActionBox2' $color={theme.colors.secondary} >
            <Text $color={theme.colors.primary}>{t('landingPage.CTA2')}</Text>
            <Vector id='R' $name='arrowLeft' $color={theme.colors.primary} $height={'4dvh'} $width={'8dvh'} $position='bottom' />
        </ActionBox>
    );
};