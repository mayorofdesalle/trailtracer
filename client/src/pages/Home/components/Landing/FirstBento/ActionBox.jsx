import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/Box';
import { slideFromRight, slideFromBottomLeft } from '@components/misc/anims';

/**
 * ActionBox
 * @description
 * This is a styled bento box that is used to contain the CTA content.
 **/
const ActionBox = styled(BentoBox)`
    grid-area: ${({ $gridArea }) => $gridArea};
    flex-direction: column;
    justify-content: space-around;
    overflow: hidden;

    & > #R {
        animation: ${slideFromRight} 1s ease-in-out;
    }

    & > #L {
        animation: ${slideFromBottomLeft} 1s ease-in-out;
    }
`;

export default ActionBox;