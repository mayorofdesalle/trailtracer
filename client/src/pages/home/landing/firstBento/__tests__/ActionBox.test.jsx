import { describe, expect, it, render, screen } from '@app/tester';

import { ActionBoxFirst, ActionBoxSecond } from '../ActionBox';

describe('ActionBoxFirst', () => {

    it('should render correctly', () => {
        render(<ActionBoxFirst data-testid='action-box-1' />);
        const actionBox = screen.getByTestId('action-box-1');

        expect(actionBox).toBeInTheDocument();
        expect(actionBox).toHaveStyle('grid-area: ActionBox1');
        expect(actionBox.firstChild).toHaveRole('img');
        expect(actionBox).toHaveTextContent(/CTA/i);
    });

});

describe('ActionBoxSecond', () => {

    it('should render correctly', () => {
        render(<ActionBoxSecond data-testid='action-box-2' />);
        const actionBox = screen.getByTestId('action-box-2');

        expect(actionBox).toBeInTheDocument();
        expect(actionBox).toHaveStyle('grid-area: ActionBox2');
        expect(actionBox.lastChild).toHaveRole('img');
        expect(actionBox).toHaveTextContent(/CTA/i);
    });

});