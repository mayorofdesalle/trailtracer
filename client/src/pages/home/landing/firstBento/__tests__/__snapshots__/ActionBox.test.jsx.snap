// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionBoxFirst > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-irLvIq kHwwsn gVwHLx kBAJVk"
  >
    <div
      class="sc-jsEeTM kWwguh left"
      role="img"
    />
    <div
      class="sc-gLLuof cFAykQ"
    >
      landingPage.CTA1
    </div>
  </div>
</DocumentFragment>
`;

exports[`ActionBoxSecond > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-irLvIq kHwwsn jPSObU diGZYn"
  >
    <div
      class="sc-gLLuof dmaRej"
    >
      landingPage.CTA2
    </div>
    <div
      class="sc-jsEeTM bBWmcy right"
      role="img"
    />
  </div>
</DocumentFragment>
`;
